operator/pkg/model/translation/translator.go

```diff
@@ -251,7 +296,78 @@ func getNamespaceNamePortsMap(m *model.Model) map[string]map[string][]string {
 				}
 				namespaceNamePortMap[be.Namespace] = namePortMap
 			}
+			mergeBackendsInNamespaceNamePortMap(r.Backends, namespaceNamePortMap)
+		}
+	}
+
+	for _, l := range m.TLS {
+		for _, r := range l.Routes {
+			mergeBackendsInNamespaceNamePortMap(r.Backends, namespaceNamePortMap)
+		}
+	}
+
+	return namespaceNamePortMap
+}
+
+// getNamespaceNamePortsMapForHTTP returns a map of namespace -> name -> ports.
+// The ports are sorted and unique.
+func getNamespaceNamePortsMapForHTTP(m *model.Model) map[string]map[string][]string {
+	namespaceNamePortMap := map[string]map[string][]string{}
+	for _, l := range m.HTTP {
+		for _, r := range l.Routes {
+			mergeBackendsInNamespaceNamePortMap(r.Backends, namespaceNamePortMap)
+		}
+	}
+	return namespaceNamePortMap
+}
+
+// getNamespaceNamePortsMapFroTLS returns a map of namespace -> name -> ports.
+// The ports are sorted and unique.
+func getNamespaceNamePortsMapForTLS(m *model.Model) map[string]map[string][]string {
+	namespaceNamePortMap := map[string]map[string][]string{}
+	for _, l := range m.TLS {
+		for _, r := range l.Routes {
+			mergeBackendsInNamespaceNamePortMap(r.Backends, namespaceNamePortMap)
 		}
 	}
 	return namespaceNamePortMap
 }
+
+func mergeBackendsInNamespaceNamePortMap(backends []model.Backend, namespaceNamePortMap map[string]map[string][]string) {
+	for _, be := range backends {
+		namePortMap, exist := namespaceNamePortMap[be.Namespace]
+		if exist {
+			namePortMap[be.Name] = sortAndUnique(append(namePortMap[be.Name], be.Port.GetPort()))
+		} else {
+			namePortMap = map[string][]string{
+				be.Name: {be.Port.GetPort()},
+			}
+		}
+		namespaceNamePortMap[be.Namespace] = namePortMap
+	}
+}
+
+func sortAndUnique(arr []string) []string {
```

Comment:
this function can be replaced by common utility added recently

https://github.com/cilium/cilium/pull/25069